lib/timers.js

```diff
@@ -229,7 +234,7 @@ function tryOnTimeout(timer, list) {
   timer._called = true;
   var threw = true;
   try {
-    timer._onTimeout();
+    ontimeout(timer);
```

Comment:
I just wrote a little test and as far as I can tell a function (or any calling functions in the call stack) shouldn't get deopted if that function *throws*.